File: Examples/CSharp/IMAP/MessagesFromIMAPServerToDisk.cs
using System;
using Aspose.Email.Clients.Imap;
using Aspose.Email.Clients;

namespace Aspose.Email.Examples.CSharp.Email.IMAP
{
    class MessagesFromIMAPServerToDisk
    {
        public static void Run()
        {
            // ExStart:MessagesFromIMAPServerToDisk
            // The path to the File directory.
            string dataDir = RunExamples.GetDataDir_IMAP();
            
            // Create an instance of the ImapClient class
            ImapClient client = new ImapClient();

            // Specify host, username, password, Port and SecurityOptions for your client
            client.Host = "imap.gmail.com";
            client.Username = "your.username@gmail.com";
            client.Password = "your.password";
            client.Port = 993;
            client.SecurityOptions = SecurityOptions.Auto;
            try
            {
                // Select the inbox folder and Get the message info collection
                client.SelectFolder(ImapFolderInfo.InBox);
                ImapMessageInfoCollection list = client.ListMessages();

                // Download each message
                for (int i = 0; i < list.Count; i++)
                {
                    // Save the EML file locally
                    client.SaveMessage(list[i].UniqueId, dataDir + list[i].UniqueId + ".eml");
                }
                // Disconnect to the remote IMAP server
                client.Dispose();
            }
            catch (Exception ex)
            {
                Console.Write(Environment.NewLine + ex);
            }
            // ExStart:MessagesFromIMAPServerToDisk
            Console.WriteLine(Environment.NewLine + "Downloaded messages from IMAP server.");
        }
    }
}

File: Examples/VisualBasic/IMAP/MessagesFromIMAPServerToDisk.vb
Imports Aspose.Email.Imap

Namespace Aspose.Email.Examples.VisualBasic.Email.IMAP
    Class MessagesFromIMAPServerToDisk
        Public Shared Sub Run()
            ' ExStart:MessagesFromIMAPServerToDisk
            ' The path to the File directory.
            Dim dataDir As String = RunExamples.GetDataDir_IMAP()

            ' Create an instance of the ImapClient class
            Dim client As New ImapClient()

            ' Specify host, username, password, Port and SecurityOptions for your client
            client.Host = "imap.gmail.com"
            client.Username = "your.username@gmail.com"
            client.Password = "your.password"
            client.Port = 993
            client.SecurityOptions = SecurityOptions.Auto
            Try
                ' Select the inbox folder and Get the message info collection
                client.SelectFolder(ImapFolderInfo.InBox)
                Dim list As ImapMessageInfoCollection = client.ListMessages()

                ' Download each message
                For i As Integer = 0 To list.Count - 1
                    ' Save the EML file locally
                    client.SaveMessage(list(i).UniqueId, dataDir + list(i).UniqueId & Convert.ToString(".eml"))
                Next
                ' Disconnect to the remote IMAP server
                client.Dispose()
            Catch ex As Exception
                Console.Write(Environment.NewLine + ex.ToString())
            End Try
            ' ExStart:MessagesFromIMAPServerToDisk
            Console.WriteLine(Environment.NewLine + "Downloaded messages from IMAP server.")
        End Sub
    End Class
End Namespace